node/tokio-utils/src/lib.rs

```diff
@@ -46,6 +48,17 @@ impl ShutdownableThread {
             let _ = thread.join().expect("Error joining child thread");
         }
     }
+
+    pub fn wait_sigint_and_shutdown(mut self) {
```

Comment:
Is `mut self` intentional? Should we explain in the docstring why do we consume `self`, is it for safety?

Reply:
doesn't really matter, actually all of the methods could consume self to make their intentions more explicit
can change this to &mut self or change others to mut self

Reply:
Whatever makes intentions more explicit.